### PR TITLE
✨ API | Add Activity Feed API

### PR DESCRIPTION
**Pull request by AntPolkanov**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ #562 

> 2. What was changed?

✏️ Added new endpoint which supports pagination.

> 3. Did you do pair or mob programming?

✏️ No

### CODE REVIEW

File: src/Application/Common/Interfaces/IActivityFeedService.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+using SSW.Rewards.Shared.DTOs.ActivityFeed;
+
+namespace SSW.Rewards.Application.Common.Interfaces;
+
+public interface IActivityFeedService
+{
+    public Task<IList<ActivityFeedViewModel>> GetActivities(ActivityFeedFilter filter, int skip, int take, CancellationToken cancellationToken);
```
Comment (matt-goldman) [inline src/Application/Common/Interfaces/IActivityFeedService.cs:7]: I don't see the value of this abstraction. I think the EF query could just be written in the handler.

File: src/Common/Shared.csproj [hunk 2]
```diff
@@ -13,6 +13,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\Domain\SSW.Rewards.Domain.csproj" />
```
Comment (matt-goldman) [inline src/Common/Shared.csproj:16]: Shared can't have a dependency on domain...but I can't see this referenced anywhere in the code so assuming this is a mistake.

**Review by matt-goldman — CHANGES_REQUESTED** [review]
Looks good to me, although we need to remove that dependency on Domain in Shared.

Re the interface and service for the activity feed, this seems unnecessary. If there's a reason for it let me know, otherwise I think the EF query should just be in the handler.

File: src/Application/Common/Interfaces/IActivityFeedService.cs (hunk 1 above)
Comment (AntPolkanov) [reply src/Application/Common/Interfaces/IActivityFeedService.cs]: Agree.

File: src/Common/Shared.csproj (hunk 2 above)
Comment (AntPolkanov) [reply src/Common/Shared.csproj]: Fixed, accidental import.

**Review by tkapa — APPROVED** [review]

**Review by matt-goldman — APPROVED** [review]
LGTM

Note that we could use automapper projections for the VM/DTO but not essential

**Pull request merged**